src/workerd/api/node/buffer.c++

```diff
@@ -120,18 +145,128 @@ kj::Array<byte> decodeHexTruncated(kj::ArrayPtr<const char> text) {
     KJ_IF_MAYBE(d1, tryFromHexDigit(text[i])) {
       b = *d1 << 4;
     } else {
+      if (strict) {
+        JSG_FAIL_REQUIRE(TypeError, "The text is not valid hex");
+      }
       break;
     }
     KJ_IF_MAYBE(d2, tryFromHexDigit(text[i+1])) {
       b |= *d2;
     } else {
+      if (strict) {
+        JSG_FAIL_REQUIRE(TypeError, "The text is not valid hex");
+      }
       break;
     }
     vec.add(b);
   }
 
   return vec.releaseAsArray();
 }
+
+uint32_t writeInto(
+    jsg::Lock& js,
+    kj::ArrayPtr<kj::byte> buffer,
+    v8::Local<v8::String> string,
+    uint32_t offset,
+    uint32_t length,
+    Encoding encoding) {
+  auto dest = buffer.slice(offset, kj::min(offset + length, buffer.size()));
+  if (dest.size() == 0 || string->Length() == 0) return 0;
```

Comment:
Not sure if our style guide calls for this, but I've seen this style lead to bugs so nit:

```suggestion
  if (dest.size() == 0 || string->Length() == 0) { return 0; }
```